zerver/data_import/rocketchat.py

```diff
@@ -0,0 +1,673 @@
+    realm_id: int,
+) -> List[ZerverFieldsT]:
+    streams = []
+
+    for rc_room_id in room_id_to_room_map:
+        channel_dict = room_id_to_room_map[rc_room_id]
+
+        now = int(timezone_now().timestamp())
+        stream_id = stream_id_mapper.get(rc_room_id)
+        invite_only = channel_dict["t"] == "p"
+
+        stream_name = channel_dict["name"]
+        stream_desc = channel_dict.get("description", "")
+        if channel_dict.get("teamId"):
+            if channel_dict.get("teamMain") is True:
+                stream_name = "[TEAM] " + stream_name
+            else:
+                stream_desc = "[Team {} channel]. {}".format(
+                    team_id_to_team_map[channel_dict["teamId"]]["name"], stream_desc
+                )
+
+        stream = build_stream(
+            date_created=now,
+            realm_id=realm_id,
+            name=stream_name,
+            description=stream_desc,
+            stream_id=stream_id,
+            deactivated=False,
+            invite_only=invite_only,
+        )
+        streams.append(stream)
+
+    return streams
+
+
+def convert_subscription_data(
+    user_id_to_user_map: Dict[str, Dict[str, Any]],
+    dsc_id_to_dsc_map: Dict[str, Dict[str, Any]],
+    zerver_stream: List[ZerverFieldsT],
+    stream_id_mapper: IdMapper,
+    user_id_mapper: IdMapper,
+    subscriber_handler: SubscriberHandler,
+) -> None:
+    stream_members_map: Dict[int, Set[int]] = {}
+
+    for rc_user_id in user_id_to_user_map:
+        user_dict = user_id_to_user_map[rc_user_id]
+
+        if not user_dict.get("__rooms"):
+            continue
+
+        for channel in user_dict["__rooms"]:
+            if channel in dsc_id_to_dsc_map:
+                # Ignore discussion rooms as these are not
+                # imported as streams, but topics.
+                continue
+            stream_id = stream_id_mapper.get(channel)
+            if stream_id not in stream_members_map:
+                stream_members_map[stream_id] = set()
+            stream_members_map[stream_id].add(user_id_mapper.get(rc_user_id))
+
+    for stream in zerver_stream:
+        if stream["id"] in stream_members_map:
+            users = stream_members_map[stream["id"]]
+        else:
+            users = set()
+        subscriber_handler.set_info(users=users, stream_id=stream["id"])
+
+
+def build_reactions(
+    total_reactions: List[ZerverFieldsT],
+    reactions: List[Dict[str, Any]],
+    message_id: int,
+    # zerver_realmemoji: List[ZerverFieldsT],
+) -> None:
+    # For the Unicode emoji codes, we use equivalent of
+    # function 'emoji_name_to_emoji_code' in 'zerver/lib/emoji' here
+    for reaction in reactions:
+        emoji_name = reaction["name"]
+        user_id = reaction["user_id"]
+        # Check in Unicode emoji
+        if emoji_name in name_to_codepoint:
+            emoji_code = name_to_codepoint[emoji_name]
+            reaction_type = Reaction.UNICODE_EMOJI
+        else:  # nocoverage
+            continue
```

Comment:
I take it we haven't implemented custom emoji import yet?

Reply:
Yes, removed all the comments related to custom emojis.